### PR TITLE
CATO-4037: Fixing validation for balance sheet statements confirmation.

### PR DESCRIPTION
**Pull request by tmikus**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8087.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+case class AC8087(value: Option[Boolean]) extends CtBoxIdentifier(name = "The directors acknowledge their responsibilities for complying with the requirements of the Act with respect to accounting records and the preparation of accounts.")
+  with CtOptionalBoolean
+  with Input
+  with ValidatableBox[Frs102AccountsBoxRetriever with FilingAttributesBoxValueRetriever]
+  with Validators {
+
+  override def validate(boxRetriever: Frs102AccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] = {
+    collectErrors(
+      failIf(boxRetriever.microEntityFiling().value)(validateAsMandatory(this))
+    )
```
Comment (pncampbell) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8087.scala]: Theres something odd here, we would never be in a micro filing if the box retriever is a 102Accounts.

**Comment by tmikus** [comment]
Wait a bit guys, have to fix a couple of things first :D

**Comment by tmikus** [comment]
It's on!

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8084.scala [hunk 2]
```diff
@@ -18,16 +18,17 @@ package uk.gov.hmrc.ct.accounts.frs102.boxes
...
   with Input
-  with ValidatableBox[Frs102AccountsBoxRetriever]
+  with ValidatableBox[Frs102AccountsBoxRetriever with FilingAttributesBoxValueRetriever]
   with Validators {
 
-  override def validate(boxRetriever: Frs102AccountsBoxRetriever): Set[CtValidation] = {
+  override def validate(boxRetriever: Frs102AccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] = {
     collectErrors(
-      validateAsMandatory(this)
+      failIf(boxRetriever.abridgedFiling().value)(validateAsMandatory(this))
```
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8084.scala]: to be consistent can we use the curly bracket notation?

Comment (tmikus) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8084.scala]: Curly bracket notation for function calls? Where do we do it that way?

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8085.scala [hunk 3]
```diff
@@ -18,16 +18,17 @@ package uk.gov.hmrc.ct.accounts.frs102.boxes
...
   with Input
-  with ValidatableBox[Frs102AccountsBoxRetriever]
+  with ValidatableBox[Frs102AccountsBoxRetriever with FilingAttributesBoxValueRetriever]
   with Validators {
 
-  override def validate(boxRetriever: Frs102AccountsBoxRetriever): Set[CtValidation] = {
+  override def validate(boxRetriever: Frs102AccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] = {
     collectErrors(
-      validateAsMandatory(this)
+      failIf(boxRetriever.abridgedFiling().value || boxRetriever.statutoryAccountsFiling().value)(validateAsMandatory(this))
```
Comment (pncampbell) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8085.scala]: You dont need the flag conditions here either (its always going to be one or the other ...)

File: src/test/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8084Spec.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      AC8084(None).validate(boxRetriever) shouldBe Set(CtValidation(Some("AC8084"), "error.AC8084.required"))
+    }
+
+    "pass validation if not empty and filing abridged accounts" in {
+      when(boxRetriever.abridgedFiling()).thenReturn(AbridgedFiling(true))
+      when(boxRetriever.statutoryAccountsFiling()).thenReturn(StatutoryAccountsFiling(false))
+
+      AC8084(Some(true)).validate(boxRetriever) shouldBe Set.empty
+    }
+
```
Comment (pncampbell) [inline src/test/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC8084Spec.scala]: Only the true+false/false+true conditions needed

**Pull request merged**